torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -254,8 +276,12 @@ static void UpdateScalarTypeForInputs(
         cast_node->addInput(input);
         cast_node->i_(attr::to, onnx_type);
         cast_node->insertBefore(n);
-        cast_node->output()->setType(CreateProfiledTensorTypeWithScalarType(
-            input_tensor_type, scalar_type));
+        if (input_tensor_type) {
+          cast_node->output()->setType(CreateProfiledTensorTypeWithScalarType(
+              input_tensor_type, scalar_type));
+        } else {
+          cast_node->output()->setType(input_list_type->getElementType());
```

Comment:
is it better to set it as a new tensortype with scalar type? otherwise the type may not be consistent with cast node.